stability: crash on nil pointer in markDead

Build sha: c382dd7


Beta cluster restarted fresh earlier today.
A few issues already encountered, including node deaths (see #5998, #6000)

`ec2-54-209-69-52.compute-1.amazonaws.com` was restarted after dying of #5998. About 5 minutes after restart, it died with:
```
W160412 13:17:12.531435 server/node.go:665  [node=2] unable to log node_restart event for node 2: failed to send RPC: too many errors encountered (2 of 2 total): rpc error: code = 4 desc = context deadline exceeded
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x896fc6]

goroutine 12 [running]:
panic(0x163ce80, 0xc82000c0a0)
        /usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/storage.(*storeDetail).markDead(0xc822b92200, 0x14449f073e5e3d6d, 0x0)
        /go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:57 +0x1b6
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
        /go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:203 +0x4f8
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82036acb0, 0xc8203d5880)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62
```

I'm intentionally including the warning as it indicates problems writing to the event log and may or may not be related.
Full log for this node since restart:
[node1.log.txt](https://github.com/cockroachdb/cockroach/files/215131/node1.log.txt)

--- a/storage/store_pool.go
+++ b/storage/store_pool.go
@@ -54,7 +54,11 @@ func (sd *storeDetail) markDead(foundDeadOn roachpb.Timestamp) {
 	sd.dead = true
 	sd.foundDeadOn = foundDeadOn
 	sd.timesDied++
-	log.Warningf("store %s on node %s is now considered offline", sd.desc.StoreID, sd.desc.Node.NodeID)
+	if sd.desc != nil {
+		// sd.desc can still be nil if it was markedAlive and enqueued in getStoreDetailLocked
+		// and never markedAlive again.
+		log.Warningf("store %s on node %s is now considered offline", sd.desc.StoreID, sd.desc.Node.NodeID)
+	}
 }
 
 // markAlive sets the storeDetail to alive(active) and saves the updated time

